indexer/etl/etl.go

```diff
@@ -0,0 +1,112 @@
+package etl
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/indexer/node"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+const (
+	defaultLoopInterval     = 5 * time.Second
+	defaultHeaderBufferSize = 500
+)
+
+type ETL struct {
+	log log.Logger
+
+	headerTraversal *node.HeaderTraversal
+	ethClient       *ethclient.Client
+	contracts       []common.Address
```

Comment:
since everyone ends up at the same place here eventually we could actually abstract right away this to 

```typescript
Array<{
name: string
abi: string
address: string
startBlock: number
}>
```
It's a safe abstraction to do since everyone always ends up here from what I've seen

Reply:
yea i definitely agree. I dont like how the abis are always extracted on every call in `processors/contracts_bedrock`. I figured we could clean this up a bit and share some contract metadata. Lets create a ticket in the backlog for this